File: src/android/app/src/main/java/com/crickettechnology/trees/Profiler.java
package com.crickettechnology.trees;

public class Profiler
{
    Profiler(String name)
    {
        this.name = name;
    }

    public void start()
    {
        this.stopwatch.reset();
        this.stopwatch.start();
    }

    public void markEnd(String label)
    {
        float elapsedTime = this.stopwatch.getElapsedTime();
        System.out.printf("%s: %.3f (%.3f) %s\n", 
                this.name, elapsedTime, elapsedTime - this.prevTime, label);
        this.prevTime = elapsedTime;
    }

    public void stop()
    {
        this.stopwatch.stop();
        System.out.printf("%s: %.3f TOTAL\n", this.name, this.stopwatch.getElapsedTime());
    }

    ////////////////////////////////////////

    private final String name;
    private float prevTime;
    private Stopwatch stopwatch = new Stopwatch();

}

File: src/ios/trees/Profiler.swift
import Foundation

class Profiler
{
    public init(name: String)
    {
        self.name = name
        self.stopwatch.reset()
        self.stopwatch.start()
    }

    public func markEnd(_ label: String)
    {
        let elapsedTime = self.stopwatch.elapsedTime
        print("\(self.name): \(formatTime(elapsedTime)) (\(formatTime(elapsedTime - self.prevTime))) \(label)")
        self.prevTime = elapsedTime
    }

    public func stop()
    {
        self.stopwatch.stop()
        print("\(self.name): \(formatTime(self.stopwatch.elapsedTime)) TOTAL")
    }

    ////////////////////////////////////////

    private let name: String
    private var prevTime: TimeInterval = 0
    private let stopwatch = Stopwatch()

    private func formatTime(_ time: TimeInterval) -> String
    {
        return String(format:"%.3f", time)
    }
}

